main/src/ca/uwaterloo/flix/api/lsp/Indexer.scala

```diff
@@ -474,13 +474,22 @@ object Indexer {
       val parent = Entity.Pattern(pat0)
       Index.occurrenceOf(pat0) ++ visitPat(pat) ++ Index.useOf(sym, loc, parent)
     case Pattern.Tuple(elms, _, _) => Index.occurrenceOf(pat0) ++ visitPats(elms)
+    case Pattern.Record(pats, pat, _, _) =>
+      Index.occurrenceOf(pat0) ++ traverse(pats)(visitRecordFieldPattern) ++ traverse(pat)(visitPat)
   }
 
   /**
     * Returns a reverse index for the given patterns `pats0`.
     */
   private def visitPats(pats0: List[Pattern]): Index = traverse(pats0)(visitPat)
 
+  /**
+    * Returns a reverse index for the given [[Pattern.Record.RecordFieldPattern]] `rfp`.
+    */
+  private def visitRecordFieldPattern(rfp: Pattern.Record.RecordFieldPattern): Index = {
+    Index.useOf(rfp.field) ++ visitType(rfp.tpe) ++ visitPat(rfp.pat)
```

Comment:
I'm not sure what this does, honestly :)

Reply:
This file is what says "a variable is used here" to the LSP so that when we click "find references" we can get it highlighted, for example.

That's why I think we need to note that a variable is declared here.

Reply:
Ok so what do I have to do?

Reply:
@mlutze Type variables don't have "definition sites" though.

Reply:
What I am getting at is that when you have `case { x } => ...` then the `x` is playing two roles: a field and a variable déclaration site. My question is how this shows up in the IDE, highlighted as a field or as a variable, and whether the indexing conflicts; what happens when I do Find references on `x`?

Reply:
Here is the current behavior (and also intended, I think): When using "go to references" on the field `x` where the record is created, it shows the pattern as a reference. When you hit "go to references" on the pattern `x`, the used variable in the match arm is shown as a reference.

![Screenshot from 2023-08-17 10-25-37](https://github.com/flix/flix/assets/60233376/5c1a651b-a8c2-4c9f-ba64-ff5a4a5b98aa)
![Screenshot from 2023-08-17 10-25-50](https://github.com/flix/flix/assets/60233376/b673b3ab-c0a0-4990-95b0-145c36ca31f2)